internal/daemonset/daemonset.go

```diff
@@ -230,6 +245,10 @@ func (dc *daemonSetGenerator) SetDevicePluginAsDesired(
 		constants.ModuleNameLabel: mod.Name,
 		constants.DaemonSetRole:   devicePluginRoleLabelValue,
 	}
+	if mod.Spec.ModuleLoader.Container.Version != "" {
```

Comment:
I am wondering if `moduleLoaderData` should implement a `moduleData` interface and then we can pass to this method (and others) a `moduleData` instead of using `Module` for devicePlugin and `modueDataLoader` for driverContainer.

Reply:
i am not sure if it worth the effort. In any case, if we decide to do it , it should be outside of this PR. You can open a Issue on that and we can discuss it there